transformer_engine/common/__init__.py

```diff
@@ -45,20 +51,32 @@ def _get_sys_extension():
     return extension
 
 
-def _load_cudnn():
-    """Load CUDNN shared library."""
-    # Attempt to locate cuDNN in Python dist-packages
-    lib_path = glob.glob(
+def _load_nvidia_cuda_library(lib_name: str):
+    """
+    Attempts to load shared object file installed via pip.
+
+    `lib_name`: Name of package as found in the `nvidia` dir in python environment.
+    """
+
+    so_paths = glob.glob(
         os.path.join(
             sysconfig.get_path("purelib"),
-            f"nvidia/cudnn/lib/libcudnn.{_get_sys_extension()}.*[0-9]",
+            f"nvidia/{lib_name}/lib/lib*.{_get_sys_extension()}.*[0-9]",
         )
     )
-    if lib_path:
-        assert (
-            len(lib_path) == 1
-        ), f"Found {len(lib_path)} libcudnn.{_get_sys_extension()}.x in nvidia-cudnn-cuXX."
-        return ctypes.CDLL(lib_path[0], mode=ctypes.RTLD_GLOBAL)
+
+    path_found = len(so_paths) > 0
+    ctypes_handles = []
+
+    if path_found:
+        for so_path in so_paths:
+            ctypes_handles.append(ctypes.CDLL(so_path, mode=ctypes.RTLD_GLOBAL))
```

Comment:
If we find multiple `.so`s, does it make sense to load all of them? Also, we should probably throw a warning if we don't find anything.

Reply:
If we don't use cuda toolkit for compilation (for JAX extensions in this PR but for everything in future PRs) then we need to load all SOs, for e.g. `libcudnn`, `libcudnn_graph`, etc.. Since this strictly checks the nvidia packages only, the SOs are never duplicate here.

I avoided throwing any errors or warnings here is because this isn't a necessary path (e.g. if `cudatoolkit` was used for compilation) so it would be disruptive and would show up on every import of TE.